pkg/networkpolicy/networkpolicy.go

```diff
@@ -0,0 +1,311 @@
+package networkpolicy
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	netv1 "k8s.io/api/networking/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NetworkPolicyBuilder provides struct for networkPolicy object.
+type NetworkPolicyBuilder struct {
+	// NetworkPolicy definition. Used to create networkPolicy object with minimum set of required elements.
+	Definition *netv1.NetworkPolicy
+	// Created networkPolicy object on the cluster.
+	Object *netv1.NetworkPolicy
+	// api client to interact with the cluster.
+	apiClient *clients.Settings
+	// errorMsg is processed before NetworkPolicy object is created.
+	errorMsg string
+}
+
+// NewNetworkPolicyBuilder method creates new instance of builder.
+func NewNetworkPolicyBuilder(
+	apiClient *clients.Settings, name, nsname string) *NetworkPolicyBuilder {
+	glog.V(100).Infof("Initializing new NetworkPolicyBuilder structure with the following params: name: %s, namespace: %s",
+		name, nsname)
+
+	builder := &NetworkPolicyBuilder{
+		apiClient: apiClient,
+		Definition: &netv1.NetworkPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the networkPolicy is empty")
+
+		builder.errorMsg = "The networkPolicy 'name' cannot be empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The namespace of the networkPolicy is empty")
+
+		builder.errorMsg = "The networkPolicy 'namespace' cannot be empty"
+	}
+
+	return builder
+}
+
+// WithNamespaceIngressRule ingress rule for the networkPolicy.
```

Comment:
Please change comment to>
// WithNamespaceIngressRule applies ingress rule for the networkPolicy.